lib/inventory.rb

```diff
@@ -41,7 +41,15 @@ def init(name, scope)
     @inventory_file_path = File.join(dir, name + ".yaml")
     if File.exist?(@inventory_file_path)
       self.merge!(Helper::CacheLoader.memo(@inventory_file_path) { |yaml|
-        YAML.load(yaml)
+        begin
+          YAML.load(yaml)
+        rescue Psych::SyntaxError
+          if restore(@inventory_file_path)
```

Comment:
__[rubocop]__
* Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. [:ref](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

<!-- Comment by [SideCI](https://sideci.com) -->